File: frontend/src/services/bookingService.js
import axiosInstance from './axiosInstance';

// Booking Service
export const bookingService = {
  // Create booking
  createBooking: async (bookingData) => {
    const response = await axiosInstance.post('/bookings', bookingData);
    return response.data;
  },

  // Get all bookings (Admin only)
  getAllBookings: async () => {
    const response = await axiosInstance.get('/bookings');
    return response.data;
  },

  // Get user's bookings
  getMyBookings: async () => {
    const response = await axiosInstance.get('/bookings/my-bookings');
    return response.data;
  },

  // Get booking by ID
  getBookingById: async (id) => {
    const response = await axiosInstance.get(`/bookings/${id}`);
    return response.data;
  },

  // Update booking status (Admin only)
  updateBookingStatus: async (id, status) => {
    const response = await axiosInstance.patch(
      `/bookings/${id}/status?status=${status}`
    );
    return response.data;
  },

  // Cancel booking
  cancelBooking: async (id) => {
    const response = await axiosInstance.delete(`/bookings/${id}`);
    return response.data;
  },
};

export default bookingService;

File: backend/src/main/java/com/drivehub/vehiclerental/service/BookingService.java
package com.drivehub.vehiclerental.service;

import com.drivehub.vehiclerental.dto.BookingRequest;
import com.drivehub.vehiclerental.dto.BookingResponse;
import com.drivehub.vehiclerental.entity.Booking.BookingStatus;

import java.util.List;

/**
 * Service interface for booking operations
 */
public interface BookingService {
    
    /**
     * Create a new booking
     * 
     * @param request Booking request
     * @param customerId Customer ID
     * @return Created booking response
     */
    BookingResponse createBooking(BookingRequest request, Long customerId);
    
    /**
     * Get booking by ID
     * 
     * @param id Booking ID
     * @return Booking response
     */
    BookingResponse getBookingById(Long id);
    
    /**
     * Get all bookings
     * 
     * @return List of booking responses
     */
    List<BookingResponse> getAllBookings();
    
    /**
     * Get bookings by customer
     * 
     * @param customerId Customer ID
     * @return List of booking responses
     */
    List<BookingResponse> getBookingsByCustomer(Long customerId);
    
    /**
     * Update booking status
     * 
     * @param id Booking ID
     * @param status New status
     * @return Updated booking response
     */
    BookingResponse updateBookingStatus(Long id, BookingStatus status);
    
    /**
     * Cancel booking
     * 
     * @param id Booking ID
     * @return Cancelled booking response
     */
    BookingResponse cancelBooking(Long id);
}
